conda/recipes/dask-cuda/meta.yaml

```diff
@@ -21,8 +21,8 @@ build:
   script:
     - {{ PYTHON }} -m pip install . -vv
   entry_points:
-    {% for e in data.get("project", {}).get("scripts", {}).items() %}
-    - {{ e|join(" = ") }}
+    {% for entrypoint in data["project"]["scripts"] %}
+    - {{ entrypoint ~ ' = ' ~ data["project"]["scripts"][entrypoint] }}
```

Comment:
I know this one looks a little weird. I can explain.

Switching from this

```
{% for e in data.get("project", {}).get("scripts", {}).items() %}
```

to this

```text
{% for e in data["project"]["scripts"].items() %}
```

results in an error like this:

```text
  File "/__w/dask-cuda/dask-cuda/conda/recipes/dask-cuda/meta.yaml", line 24, in top-level template code
    {% for e in data["project"]["scripts"].items() %}
TypeError: 'NoneType' object is not callable
```

([build link](https://github.com/rapidsai/dask-cuda/actions/runs/9485331325/job/26137086269))

It seems that calling dictionary methods like `.items()` or `.iteritems()` isn't supported... maybe `conda-build` overrides `.get()` or `__getattr__` or `getattr()`, or maybe that's something `Jinja2` is doing.

BUT... this pattern where you `for`-iterate over **keys**, and then use those to subset the list, totally works 😁 

And it does come with a bit more safety. I tried removing the `[project.scripts]` table in `pyproject.toml` completely. On `branch-24.08`, that results in the recipe silently building "successfully" without those entrypoints defined.

On this branch, it results in an informative error.

```text
Error: Failed to render jinja template in /Users/jlamb/repos/dask-cuda/conda/recipes/dask-cuda/meta.yaml:
'dict object' has no attribute 'scripts'
```

See "How I tested this" in the PR description for details on how I tested this.

Reply:
"I can explain, I promise!" is always a good thing to write on a diff. 😉 I think this is totally fine.

Reply:
haha yeah, thanks 😂